applications/StructuralMechanicsApplication/tests/constitutive_law_test.py

```diff
@@ -93,33 +61,127 @@ def test_Uniaxial_HyperElastic_3D(self):
         cl_params.CheckAllParameters() # Can not use this until the geometry is correctly exported to python
         cl_params.CheckMechanicalVariables()
         cl_params.CheckShapeFunctions()
+        
+        return cl_params
+    
+    def _cl_check(self, cl, properties, geom, model_part, dim):
+        cl.Check(properties, geom, model_part.ProcessInfo)
 
-        #print("The Material Response PK2")
-        #cl.CalculateMaterialResponsePK2(cl_params)
-        #print("Stress = ", cl_params.GetStressVector())
-        #print("Strain = ", cl_params.GetStrainVector())
-        #print("C      = ", cl_params.GetConstitutiveMatrix())
+        if(cl.WorkingSpaceDimension() != dim):
+            raise Exception("Mismatch between the WorkingSpaceDimension of the Constitutive Law and the dimension of the space in which the test is performed")
+    
+    def _create_F(self):
+        F = KratosMultiphysics.Matrix(3,3)
+        F[0,0] = 1.0; F[0,1] = 0.0; F[0,2] = 0.0;
+        F[1,0] = 0.0; F[1,1] = 1.0; F[1,2] = 0.0;
+        F[2,0] = 0.0; F[2,1] = 0.0; F[2,2] = 1.0;
+        return F
+    
+    def _set_cl_options(self, dict_options):
```

Comment:
why are u doing this?

Reply:
What do you mean?

Reply:
why using a dictionary i mean.

Reply:
Because there are many options and dictionaries are useful

Reply:
The CO already has an internal dictionary of options. I Don t understand why you would like to replicate that

Reply:
Because this way I now the options I have and what I can set

Reply:
i find this a rather weak argument. The list of options of the CL is documented in the wiki and should not change.

i really don't see how 

     cl.Set(USE_ELEMENT_PROVIDED_STRAIN, False) 

is worst than

     dict_options = {'USE_ELEMENT_PROVIDED_STRAIN': False, 

still i will accept this to end a sterile discussion